src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/IslRulesService.java

```diff
@@ -0,0 +1,131 @@
+import org.openkilda.floodlight.api.request.rulemanager.MeterCommand;
+import org.openkilda.floodlight.api.request.rulemanager.OfCommand;
+import org.openkilda.floodlight.api.response.rulemanager.SpeakerCommandResponse;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.rulemanager.DataAdapter;
+import org.openkilda.rulemanager.FlowSpeakerData;
+import org.openkilda.rulemanager.GroupSpeakerData;
+import org.openkilda.rulemanager.MeterSpeakerData;
+import org.openkilda.rulemanager.RuleManager;
+import org.openkilda.rulemanager.SpeakerData;
+import org.openkilda.rulemanager.adapter.PersistenceDataAdapter;
+import org.openkilda.wfm.share.model.Endpoint;
+import org.openkilda.wfm.share.model.IslReference;
+
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class IslRulesService {
+    private final IIslCarrier carrier;
+    private final PersistenceManager persistenceManager;
+    private final RuleManager ruleManager;
+
+    private final Map<UUID, Request> installRequests = new HashMap<>();
+    private final Map<UUID, Request> deleteRequests = new HashMap<>();
+
+    public IslRulesService(IIslCarrier carrier, PersistenceManager persistenceManager, RuleManager ruleManager) {
+        this.carrier = carrier;
+        this.persistenceManager = persistenceManager;
+        this.ruleManager = ruleManager;
+    }
+
+    /**
+     * Build and send required ISL service rules install commands.
+     */
+    public void sendInstallIslServiceRules(IslReference reference, Endpoint endpoint) {
+        List<OfCommand> commands = buildIslServiceRules(endpoint.getDatapath(), endpoint.getPortNumber());
+        UUID commandId = UUID.randomUUID();
+        installRequests.put(commandId, new Request(reference, endpoint));
+        carrier.sendIslDefaultRulesInstallCommand(endpoint.getDatapath(), commandId, commands);
+    }
+
+    /**
+     * Build and send required ISL service rules delete commands.
+     */
+    public void sendDeleteIslServiceRules(IslReference reference, Endpoint endpoint) {
+        List<OfCommand> commands = buildIslServiceRules(endpoint.getDatapath(), endpoint.getPortNumber());
+        UUID commandId = UUID.randomUUID();
+        deleteRequests.put(commandId, new Request(reference, endpoint));
+        carrier.sendIslDefaultRulesDeleteCommand(endpoint.getDatapath(), commandId, commands);
+    }
+
+    /**
+     * Process ISL service rules installed/deleted response.
+     */
+    public void handleResponse(SpeakerCommandResponse response) {
+        Request request = installRequests.remove(response.getCommandId());
+        if (request != null) {
+            if (response.isSuccess()) {
+                carrier.processIslRuleInstalled(request.getReference(), request.getEndpoint());
```

Comment:
Do we really need install/delete request separation everiwhere here? Threy are processed in same way, the only difference is some wrapper types and some specific methos calls (with same set of argumetns). Is there a way to merge this code?